src/pages/SmartContractInteraction/components/UpgradeSection.tsx

```diff
@@ -48,75 +52,70 @@ export const UpgradeSection = ({ chain }: { chain: EnvironmentsEnum }) => {
-            <input
-              onChange={onUpload}
-              className={styles.field}
-              id='upgrade_file_input'
-              type='file'
-              accept='.wasm'
-            />
-          </div>
-          <div className={styles.upload}>
-            <label htmlFor='contract_address' className={styles.label}>
-              Contract Address
-            </label>
-            <input
-              id='contract_address'
-              type='text'
-              className={styles.field}
-              autoComplete='off'
-              value={upgradeContractAddress}
-              onChange={(e) => setUpgradeContractAddress(e.target.value)}
-            />
-          </div>
-          {!upgradeContractAddress && (
-            <div className={styles.error}>
-              <span>Contract address is required</span>
-            </div>
-          )}
-
-          <div className={styles.buttons}>
-            <button
-              onClick={() =>
-                isLoggedIn ? handleUpgrade() : setShowProvidersModal(true)
-              }
-              className={styles.button}
-              disabled={!wasmCode || !upgradeContractAddress}
-            >
-              Upgrade
-            </button>
-          </div>
-          <textarea
-            rows={10}
+    <div className={styles.smartcontract}>
+      <div className={styles.form}>
+        <div className={styles.upload}>
+          <label htmlFor='upgrade_file_input' className={styles.label}>
+            Upload .wasm file
+          </label>
+          <input
+            onChange={onUpload}
             className={styles.field}
-            placeholder='.wasm code will be displayed here...'
-            value={wasmCode?.toString()}
-            readOnly={true}
+            id='upgrade_file_input'
+            type='file'
+            accept='.wasm'
           />
-          {contractOrDeployerAddress && (
-            <div className={styles.result}>
-              <strong>Deployer Address:</strong>
-              <Trim className={styles.value} text={contractOrDeployerAddress} />
-              <CopyButton
-                text={contractOrDeployerAddress}
-                className={styles.copy}
-              />
-            </div>
-          )}
         </div>
+        <div className={styles.upload}>
+          <label htmlFor='contract_address' className={styles.label}>
+            Contract Address
+          </label>
+          <input
+            id='contract_address'
+            type='text'
+            className={styles.field}
+            autoComplete='off'
+            value={upgradeContractAddress}
+            onChange={(e) => setUpgradeContractAddress(e.target.value)}
+          />
+        </div>
+        {!upgradeContractAddress && (
+          <div className={styles.error}>
+            <span>Contract address is required</span>
+          </div>
+        )}
+
+        <div className={styles.buttons}>
+          <button
+            onClick={() =>
```

Comment:
The inline callback should be declared in a variable and have the conditions written there. 👍